rust/arrow/src/ipc/file/reader.rs

```diff
@@ -328,11 +351,38 @@ fn create_list_array(
     }
 }
 
+/// Reads the correct number of buffers based on list type an null_count, and creates a
```

Comment:
This reflects the origins of Dictionary in List, its closest counterpart. We will need a more general access mechanism for dictionaries such as a singleton data type so that value() will work. Of course in practice, this will always be inefficient and it is better to turn slices of arrays into rust arrays. We should add an item to do this going forward. At present I'm using a single value ArrayRef to get us there.